protocol/v2/ssv/validator/non_committee_validator.go

```diff
@@ -161,53 +147,51 @@ func (ncv *CommitteeObserver) ProcessMessage(msg *queue.SSVMessage) error {
 			}
 
 			logger.Info("✅ saved participants",
-				zap.String("converted_role", beaconRole.ToBeaconRole()),
+				zap.String("role", beaconRole.String()),
 				zap.Uint64("validator_index", uint64(key.ValidatorIndex)),
 				fields.Validator(validator.ValidatorPubKey[:]),
 				zap.String("signers", strings.Join(operatorIDs, ", ")),
-				zap.String("msg_id", hex.EncodeToString(msgID[:])),
+				// zap.String("msg_id", hex.EncodeToString(msgID[:])), TODO replace with?
 				fields.BlockRoot(key.Root),
 			)
 
 			if ncv.newDecidedHandler != nil {
-				ncv.newDecidedHandler(qbftstorage.ParticipantsRangeEntry{
-					Slot:       slot,
-					Signers:    quorum,
-					Identifier: msgID,
-				})
+				p := qbftstorage.Participation{
+					ParticipantsRangeEntry: qbftstorage.ParticipantsRangeEntry{
+						Slot:    slot,
+						Signers: quorum,
+					},
+					DomainType: spectypes.DomainType(msg.MsgID.GetDomain()),
+					Role:       beaconRole,
+					PK:         validator.ValidatorPubKey,
+				}
+
+				ncv.newDecidedHandler(p)
 			}
 		}
 	}
 
 	return nil
 }
 
-func (ncv *CommitteeObserver) getBeaconRoles(msg *queue.SSVMessage, root phase0.Root) []convert.RunnerRole {
+func (ncv *CommitteeObserver) getBeaconRoles(msg *queue.SSVMessage, root phase0.Root) []spectypes.BeaconRole {
 	if msg.MsgID.GetRoleType() == spectypes.RoleCommittee {
 		attester := ncv.attesterRoots.Get(root)
 		syncCommittee := ncv.syncCommRoots.Get(root)
 
 		switch {
 		case attester != nil && syncCommittee != nil:
-			return []convert.RunnerRole{convert.RoleAttester, convert.RoleSyncCommittee}
+			return []spectypes.BeaconRole{spectypes.BNRoleAttester, spectypes.BNRoleSyncCommittee}
 		case attester != nil:
-			return []convert.RunnerRole{convert.RoleAttester}
+			return []spectypes.BeaconRole{spectypes.BNRoleAttester}
 		case syncCommittee != nil:
-			return []convert.RunnerRole{convert.RoleSyncCommittee}
+			return []spectypes.BeaconRole{spectypes.BNRoleSyncCommittee}
 		default:
 			return nil
 		}
 	}
-	return []convert.RunnerRole{casts.RunnerRoleToConvertRole(msg.MsgID.GetRoleType())}
-}
-
-// nonCommitteeInstanceContainerCapacity returns the capacity of InstanceContainer for non-committee validators
-func nonCommitteeInstanceContainerCapacity(fullNode bool) int {
-	if fullNode {
-		// Helps full nodes reduce
-		return 2
-	}
-	return 1
+	// nolint: gosec
+	return []spectypes.BeaconRole{spectypes.BeaconRole(msg.MsgID.GetRoleType())}
```

Comment:
this would no longer work because BeaconRole and RunnerRole are incompatible

a switch should be done instead to map RunnerRole to BeaconRole, where the only special case would be the RoleCommittee which is already implemented above